controllers/ControllerComment.php

```diff
@@ -2,100 +2,100 @@
 
 namespace OC_Blog\Controllers;
 
-use GuzzleHttp\Psr7\ServerRequest;
-use OC_Blog\Config\ConstantGlobal;
 use OC_Blog\Models\CommentsManager;
+use OC_Blog\Tools\ControllerFactory;
 use OC_Blog\Tools\Session;
 
 
-class ControllerComment {
+class ControllerComment extends ControllerFactory {
 
-	private array $_params;
-	private array $_method;
-	private object $_twig;
-	private object $_commentManager;
-	private string $_server;
-
-
-	public function __construct($method, $twig, $params){
-		$this->_method = $method;
-		$this->_twig = $twig;
-		$this->_params = $params;
-		$this->_commentManager = new CommentsManager();
-		$this->_server = ( new ConstantGlobal(ServerRequest::fromGlobals()) )->getServerName()['SERVER_NAME'];
-		$target = $method[2];
-		if (method_exists(ControllerComment::class, $target) ) {
-			$this->$target();
-		}else{
-			echo $this->_twig->render('404.twig');
-		}
-	}
-
-	public function addCommentPost(){
+	/**
+	 *Ajoute un commentaire à post en BDD.
+	 *
+	 * Si l'utilisateur n'est pas connecté, renvoi sur la page de connexion.
+	 */
+	public function addCommentPost(): void {
 		$key = (new Session)->getKey('user');
 
 
 		if (!isset($key)){
-			header("Location: http://".$this->_server."/Auth/login");
+			header("Location: http://".$this->getServer()."/Auth/login");
```

Comment:
Codacy found an issue: [The use of function header() is discouraged](https://app.codacy.com/manual/PatOpen/OC_Blog/pullRequest?prid=6061935)